src/Widget/Field.php

```diff
@@ -936,7 +936,11 @@ private function addSuccessCssClassToInput(): void
         if ($this->validationStateOn === 'input') {
             $attributeName = HtmlForm::getAttributeName($this->attribute);
 
-            if (!$this->data->hasErrors($attributeName)) {
+            if (
+                !$this->data->hasErrors($attributeName) &&
+                $this->data->isValidated() &&
+                !empty($this->data->getAttributeValue($attributeName))
```

Comment:
```suggestion
                $this->data->isValidated() &&
                !$this->data->hasErrors($attributeName) &&
                !empty($this->data->getAttributeValue($attributeName))
```

Reply:
`!empty($this->data->getAttributeValue($attributeName))` condition looks odd. What about "should be empty" or "optional value" validation?

Reply:
not simply that it will have fields that do not have any validation and will be empty, so I think we should not put the has-success class, or by default if it does not have any validation put has-success on it, what do you think?